src/bidiMapper/modules/cdp/CdpTarget.ts

```diff
@@ -134,8 +144,20 @@ export class CdpTarget {
   async #unblock() {
     try {
       await Promise.all([
-        this.#cdpClient.sendCommand('Runtime.enable'),
         this.#cdpClient.sendCommand('Page.enable'),
+        // There can be some existing frames in the target, if reconnecting to an
+        // existing browser instance, e.g. via Puppeteer. Need to restore the browsing
+        // contexts for the frames to correctly handle further events, like
+        // `Runtime.executionContextCreated`.
+        // It's important to schedule this task together with enabling domains commands to
+        // prepare the tree before the events (e.g. Runtime.executionContextCreated) start
+        // coming.
+        this.#cdpClient
+          .sendCommand('Page.getFrameTree')
```

Comment:
perhaps we could add a test to make sure that evaluate works right after a context creation (the puppeteer job does not report that scenario as fixed yet)

Reply:
The puppeteer tests started passing:
> **Remove** the following from the TestExpectations.json to ignore the error:
> * [launcher.spec] Launcher specs Puppeteer Puppeteer.connect should be able to reconnect to a disconnected browser"
> * [launcher.spec] Launcher specs Puppeteer Puppeteer.connect should be able to reconnect"